iconservice/meta/storage.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2019 ICON Foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import TYPE_CHECKING, Optional
+
+from ..database.db import ExternalDatabase
+from ..icon_constant import IconScoreContextType
+from ..utils.msgpack_for_db import MsgPackForDB
+
+if TYPE_CHECKING:
+    from ..iconscore.icon_score_context import IconScoreContext
+    from ..database.batch import ExternalBatch
+
+
+class Storage(object):
+    _KEY_LAST_CALC_END_BLOCK = b'last_calc_end_block'
+    _KEY_LAST_TERM_END_BLOCK = b'last_term_end_block'
+
+    def __init__(self):
+        self._db: Optional['ExternalDatabase'] = None
+
+    def open(self, path: str):
+        self._db = ExternalDatabase.from_path(path, create_if_missing=True)
+
+    def close(self):
+        """Close the embedded database.
+        """
+        if self._db:
+            self._db.close()
+            self._db = None
+
+    def put_last_calc_end_block(self,
+                                batch: 'ExternalBatch',
+                                last_end_block_height: int):
+        version = 0
+        value: bytes = MsgPackForDB.dumps([version, last_end_block_height])
+        batch[self._KEY_LAST_CALC_END_BLOCK] = value
+
+    def get_last_calc_end_block(self,
+                                context: 'IconScoreContext') -> int:
```

Comment:
'return' value could be  'None' when starting of the first calculation (i.e. starting of IISS_REV)